libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/navigation/speedlimit/CarSpeedLimitRenderer.kt

```diff
@@ -46,39 +52,52 @@ class CarSpeedLimitRenderer(
         }
     }
 
+    private var scope: CoroutineScope? = null
+
     private fun updateSpeed(locationMatcherResult: LocationMatcherResult) {
         val speedKmph =
             locationMatcherResult.enhancedLocation.speed / METERS_IN_KILOMETER * SECONDS_IN_HOUR
+        val speedLimitOptions = mainCarContext.speedLimitOptions.value
+        val signFormat = speedLimitOptions.forcedSignFormat
+            ?: locationMatcherResult.speedLimit?.speedLimitSign
+        val threshold = speedLimitOptions.warningThreshold
         when (mainCarContext.mapboxNavigation.navigationOptions.distanceFormatterOptions.unitType) {
             UnitType.IMPERIAL -> {
                 val speedLimit =
                     locationMatcherResult.speedLimit?.speedKmph?.let { speedLimitKmph ->
                         5 * (speedLimitKmph / KILOMETERS_IN_MILE / 5).roundToInt()
                     }
                 val speed = speedKmph / KILOMETERS_IN_MILE
-                speedLimitWidget?.update(speedLimit, speed.roundToInt())
+                speedLimitWidget?.update(speedLimit, speed.roundToInt(), signFormat, threshold)
             }
             UnitType.METRIC -> {
-                speedLimitWidget?.update(
-                    locationMatcherResult.speedLimit?.speedKmph,
-                    speedKmph.roundToInt()
-                )
+                val speedLimit = locationMatcherResult.speedLimit?.speedKmph
+                speedLimitWidget?.update(speedLimit, speedKmph.roundToInt(), signFormat, threshold)
             }
         }
     }
 
     override fun onAttached(mapboxCarMapSurface: MapboxCarMapSurface) {
         logAndroidAuto("CarSpeedLimitRenderer carMapSurface loaded")
-        val speedLimitWidget = SpeedLimitWidget().also { speedLimitWidget = it }
+        val forcedSignFormat = mainCarContext.speedLimitOptions.value.forcedSignFormat
+        val speedLimitWidget = SpeedLimitWidget(
+            initialSignFormat = forcedSignFormat ?: SpeedLimitSign.MUTCD,
+        ).also { speedLimitWidget = it }
         mapboxCarMapSurface.mapSurface.addWidget(speedLimitWidget)
         MapboxNavigationApp.registerObserver(navigationObserver)
+        val scope = MainScope().also { scope = it }
```

Comment:
```suggestion
        scope = MainScope()
```

a little cleaner if using lateinit